### PR TITLE
add inception_v3.py

### PR DESCRIPTION
**Pull request by FredHuang16**
add inception_v3

### CODE REVIEW

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 1]
```diff
@@ -0,0 +1,502 @@
+# copyright (c) 2020 PaddlePaddle Authors. All Rights Reserve.
```
Comment (RainFrost1) [inline ppcls/arch/backbone/legendary_models/inception_v3.py]: 2021

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+__all__ = ["InceptionV3"]
+
+
+NET_CONFIG = {
+    'inception_a':[[192, 256, 288], [32, 64, 64]],
+    'inception_b':[288],   
+    'inception_c':[[768, 768, 768, 768], [128, 160, 160, 192]],
+    'inception_d':[768],   
+    'inception_e':[1280,2048]
+}
```
Comment (RainFrost1) [inline ppcls/arch/backbone/legendary_models/inception_v3.py:47]: 加一下config的注释吧

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+class ConvBNLayer(TheseusLayer):
+    def __init__(self,
+                 num_channels,
+                 num_filters,
+                 filter_size,
+                 stride=1,
+                 padding=0,
+                 groups=1,
+                 act="relu",
+                 name=None):
```
Comment (RainFrost1) [inline ppcls/arch/backbone/legendary_models/inception_v3.py]: `name`参数在这一版中全都去掉了。之前版本`name`主要是对齐静态图的参数，现在已经不需要了

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+        super(ConvBNLayer, self).__init__()
+
+        self.conv = Conv2D(
+            in_channels=num_channels,
+            out_channels=num_filters,
+            kernel_size=filter_size,
+            stride=stride,
+            padding=padding,
+            groups=groups,
+            weight_attr=ParamAttr(name=name+"_weights"),
```
Comment (RainFrost1) [inline ppcls/arch/backbone/legendary_models/inception_v3.py]: 删除`weight_attr` name，下同`param_attr``bias_attr``mean_name``variance_name`等

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 5]
```diff
@@ -0,0 +1,502 @@
...
+
+class InceptionStem(TheseusLayer):
+    def __init__(self):
+        super(InceptionStem, self).__init__()
+        self.conv_1a_3x3 = ConvBNLayer(num_channels=3,
+                                       num_filters=32,
+                                       filter_size=3,
+                                       stride=2,
+                                       act="relu",
+                                       name="conv_1a_3x3")
```
Comment (RainFrost1) [inline ppcls/arch/backbone/legendary_models/inception_v3.py]: `name`同上

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 6]
```diff
@@ -0,0 +1,502 @@
...
+
+        self.inception_a_list = config['inception_a']
+        self.inception_c_list = config['inception_c']
+        self.inception_b_list = config['inception_b']
+        self.inception_d_list = config['inception_d']
+        self.inception_e_list = config ['inception_e']
+        
+        self.inception_stem = InceptionStem()
+
+        self.inception_block_list = []
```
Comment (RainFrost1) [inline ppcls/arch/backbone/legendary_models/inception_v3.py]: 使用`nn.Sequential`或者`nn.LayerList`，这样避免`name`的定义。下同

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+
+import paddle
+from paddle import ParamAttr
+import paddle.nn as nn
+import paddle.nn.functional as F
+from paddle.nn import Conv2D, BatchNorm, Linear, Dropout
+from paddle.nn import AdaptiveAvgPool2D, MaxPool2D, AvgPool2D
+from paddle.nn.initializer import Uniform
+import math
+
```
Comment (cuicheng01) [inline ppcls/arch/backbone/legendary_models/inception_v3.py:24]: 这里加上from ppcls.utils.save_load import load_dygraph_pretrain吧，在下边load权重

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+        self.conv = Conv2D(
+            in_channels=num_channels,
+            out_channels=num_filters,
+            kernel_size=filter_size,
+            stride=stride,
+            padding=padding,
+            groups=groups,
+            bias_attr=False)
+        self.batch_norm = BatchNorm(
+            num_filters,
```
Comment (cuicheng01) [inline ppcls/arch/backbone/legendary_models/inception_v3.py]: act和bn分开写吧

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 9]
```diff
@@ -0,0 +1,453 @@
...
+        y = self.conv_2a_3x3(y)
+        y = self.conv_2b_3x3(y)
+        y = self.maxpool(y)
+        y = self.conv_3b_1x1(y)
+        y = self.conv_4a_3x3(y)
+        y = self.maxpool(y)
+        return y
+
+                         
+class InceptionA(TheseusLayer):
```
Comment (cuicheng01) [inline ppcls/arch/backbone/legendary_models/inception_v3.py:118]: 这部分重复比较多，InceptionA模块中是不都可以用nn.Sequential串起来呢，BCDE同理

File: ppcls/arch/backbone/legendary_models/inception_v3.py (hunk 2 above)
Comment (FredHuang16) [reply ppcls/arch/backbone/legendary_models/inception_v3.py:47]: 已加

File: ppcls/arch/backbone/legendary_models/inception_v3.py (hunk 7 above)
Comment (FredHuang16) [reply ppcls/arch/backbone/legendary_models/inception_v3.py:24]: 已加

File: ppcls/arch/backbone/legendary_models/inception_v3.py (hunk 8 above)
Comment (FredHuang16) [reply ppcls/arch/backbone/legendary_models/inception_v3.py]: 还好，也没什么问题

File: ppcls/arch/backbone/legendary_models/inception_v3.py (hunk 9 above)
Comment (FredHuang16) [reply ppcls/arch/backbone/legendary_models/inception_v3.py:118]: 还好，也没有什么问题

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 10]
```diff
@@ -0,0 +1,494 @@
...
+
+        for i in range(len(self.inception_d_list)):
+            inception_d = InceptionD(self.inception_d_list[i])
+            self.inception_block_list.append(inception_d)
+
+        for i in range(len(self.inception_e_list)):
+            inception_e = InceptionE(self.inception_e_list[i])
+            self.inception_block_list.append(inception_e)
+ 
+        self.gap = AdaptiveAvgPool2D(1)
```
Comment (RainFrost1) [inline ppcls/arch/backbone/legendary_models/inception_v3.py]: `self.avg_pool`，基本的Layer咱们最好统一一下

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 11]
```diff
@@ -0,0 +1,494 @@
...
+        self.drop = Dropout(p=0.2, mode="downscale_in_infer")
```
Comment (RainFrost1) [inline ppcls/arch/backbone/legendary_models/inception_v3.py]: `self.dropout`

File: ppcls/arch/backbone/legendary_models/inception_v3.py [hunk 12]
```diff
@@ -0,0 +1,494 @@
...
+        stdv = 1.0 / math.sqrt(2048 * 1.0)
+        self.out = Linear(
```
Comment (RainFrost1) [inline ppcls/arch/backbone/legendary_models/inception_v3.py]: `self.fc`

File: ppcls/arch/backbone/legendary_models/inception_v3.py (hunk 11 above)
Comment (FredHuang16) [reply ppcls/arch/backbone/legendary_models/inception_v3.py]: 已改

File: ppcls/arch/backbone/legendary_models/inception_v3.py (hunk 10 above)
Comment (FredHuang16) [reply ppcls/arch/backbone/legendary_models/inception_v3.py]: 已改

File: ppcls/arch/backbone/legendary_models/inception_v3.py (hunk 12 above)
Comment (FredHuang16) [reply ppcls/arch/backbone/legendary_models/inception_v3.py]: 已改

**Review by RainFrost1 — APPROVED** [review]
